src/main/java/hudson/plugins/ansicolor/AnsiColorConsoleLogFilter.java

```diff
@@ -20,10 +24,40 @@
     private static final Logger LOG = Logger.getLogger(AnsiColorConsoleLogFilter.class.getName());
 
     private AnsiColorMap colorMap;
+    private final Map<String, byte[]> notes;
 
     public AnsiColorConsoleLogFilter(AnsiColorMap colorMap) {
         super();
         this.colorMap = colorMap;
+        this.notes = new HashMap<>();
+        // some cases of AnsiHtmlOutputStream.setForegroundColor:
+        for (AnsiColorMap.Color color : AnsiColorMap.Color.values()) {
+            pregenerateNote(new AnsiAttributeElement(AnsiAttributeElement.AnsiAttrType.FG, "span", "style=\"color: " + colorMap.getNormal(color.ordinal()) + ";\""));
+        }
+        // TODO other cases, and other methods
```

Comment:
The tedious part. Handling bright colors, `swapColors`, bold/italic/underline, etc. Basically every call to the `AnsiAttributeElement` constructor from `AnsiHtmlOutputStream` which could be predicted given an `AnsiColorMap`—probably all but the explicit use of RGB coördinates, which I am guessing are rarely used anyway.